### PR TITLE
[Improvement] Update the generating paral config 

### PR DESCRIPTION
**Pull request by xin-chen42**
### What changes were proposed in this pull request?

Add a strategy for adjusting learning_rate and weight_decay when batch_size is modified.

### Why are the changes needed?

When batch_size is modified, it can affect the training performance. Adjusting the learning_rate and weight_decay can help mitigate this impact.

### Does this PR introduce any user-facing change?

None.

### How was this patch tested?

Use test_strategy_generator.py.

### CODE REVIEW

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 1]
```diff
@@ -115,18 +118,35 @@ def _generate_dataloader_config(
...
+            except ZeroDivisionError:
+                ratio = 1
+            coefficient = math.sqrt(ratio)
+
+            update_version = optimizer_config.version + 1
+            update_learning_rate = optimizer_config.learning_rate * coefficient
+            update_weight_decay = optimizer_config.weight_decay * coefficient
+
+            logger.info(f"update_learning_rate: {update_learning_rate}")
+            logger.info(f"update_weight_decay: {update_weight_decay}")
```
Comment (workingloong) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:140]: `logger.info(f"Update optimizer with learning rate {update_learning_rate} and weight decay {update_weight_decay}")`

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 2]
```diff
@@ -73,11 +76,11 @@ def generate_opt_strategy(
             logger.info(f"paral_configs: {paral_configs}")
             return paral_configs[0]
 
-    def _generate_dataloader_config(
```
Comment (workingloong) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:76]: Do not merge the methods to generate the dataloader and optimizer config. Otherwise, the method is too long.

File: dlrover/python/common/grpc.py [hunk 3]
```diff
@@ -386,6 +386,7 @@ class DataLoaderConfig(Message):
 
     version: int = 0
     dataloader_name: str = ""
+    last_batch_size: int = 0
```
Comment (merlintang) [inline dlrover/python/common/grpc.py:389]: it does not make sense to add this para into this grpc file

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 4]
```diff
@@ -118,15 +125,36 @@ def _generate_dataloader_config(
...
+        last_batch_size = dataloader_config.last_batch_size
+
+        try:
+            ratio = batch_size / last_batch_size
+        except ZeroDivisionError:
+            ratio = 1
+        coefficient = math.sqrt(ratio)
+
+        update_version = optimizer_config.version + 1
+        update_learning_rate = optimizer_config.learning_rate * coefficient
```
Comment (merlintang) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:145]: write down the math formual here to illustrate this procedure. 
// learn-rate = learn-rate * xxx

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 5]
```diff
@@ -118,15 +122,45 @@ def _generate_dataloader_config(
...
+        # and the previous batch_size
+        try:
+            ratio = batch_size / last_batch_size
+        except ZeroDivisionError:
+            ratio = 1
+        coefficient = math.sqrt(ratio)
+
+        update_version = optimizer_config.version + 1
+
+        # update_learning_rate = learning_rate * sqrt(ratio)
```
Comment (workingloong) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:145]: Remove the unused code.

**Review by workingloong — APPROVED** [review]
LGTM
